src/pages/explore.tsx

```diff
@@ -1,554 +1,520 @@
+    <>
+      <InlineField label="Query" grow={true}>
+        <Input
+          defaultValue={queryStringVariable.getValue().toString()}
+          placeholder="Lucene Query"
+          onKeyDown={e => (e.key === 'Enter' ? model.doQuery() : null)}
+          onChange={e => model.onTextChange(e.currentTarget.value)}
+        />
+      </InlineField>
+      {timeseriesLoading || logsLoading ? (
+        <Button
+          icon="fa fa-spinner"
+          onClick={() => {
+            logsQueryRunner.cancelQuery();
+            histogramQueryRunner.cancelQuery();
+          }}
+          variant="destructive"
+        >
+          Cancel
+        </Button>
+      ) : (
+        <Button icon="sync" onClick={model.doQuery}>
+          Run Query
+        </Button>
+      )}
+    </>
+  );
 };
 
-const getShiftedTimeRange = (direction: number, origRange: TimeRange): TimeRange => {
-  const range = {
-    from: toUtc(origRange.from),
-    to: toUtc(origRange.to),
-  };
+class KaldbQuery extends SceneObjectBase<KaldbQueryState> {
+  static Component = KaldbQueryRenderer;
 
-  const timespan = (range.to.valueOf() - range.from.valueOf()) / 2;
-  let to: number, from: number;
-
-  if (direction === -1) {
-    to = range.to.valueOf() - timespan;
-    from = range.from.valueOf() - timespan;
-  } else if (direction === 1) {
-    to = range.to.valueOf() + timespan;
-    from = range.from.valueOf() + timespan;
-    if (to > Date.now() && range.to.valueOf() < Date.now()) {
-      to = Date.now();
-      from = range.from.valueOf();
-    }
-  } else {
-    to = range.to.valueOf();
-    from = range.from.valueOf();
+  constructor(state?: Partial<KaldbQueryState>) {
+    super({
+      query: '*:*',
+      timeseriesLoading: false,
+      logsLoading: false,
+      ...state,
+    });
   }
 
-  return rangeUtil.convertRawToRange({ from: dateTimeParse(from), to: dateTimeParse(to) });
-};
-
-export const Explore: FC<AppRootProps> = ({ query, path, meta }) => {
-  const [intState, setIntState] = useState({
-    queryString: query.queryString ? query.queryString : '',
-    dataSource: query.dataSource
-      ? query.dataSource
-      : getDataSourceSrv().getList({
-          type: ['slack-kaldb-app-backend-datasource'],
-        })[0].name,
-    timeRange:
-      query.from && query.to
-        ? rangeUtil.convertRawToRange({
-            from: isNaN(query.from) ? query.from : dateTimeParse(parseInt(query.from, 10)),
-            to: isNaN(query.to) ? query.to : dateTimeParse(parseInt(query.to, 10)),
-          })
-        : getDefaultTimeRange(),
-  });
+  doQuery = () => {
+    queryStringVariable.setValue(this.state.query);
+  };
 
-  const setSynchronizedState = (state: any) => {
-    // merge input state with existing state, overwriting where necessary
-    setIntState(Object.assign(intState, state));
-    syncStateToUrl();
+  onTextChange = (query: string) => {
+    if (query.length === 0) {
+      this.setState({
+        query: '*:*',
+      });
+    } else {
```

Comment:
This was only for the internal state, and never populated the user visible field with a `*:*`. However I went ahead and entirely removed the `*:*` from the code, as this will be automatically handled by the datasource if an empty string is provided.